awslambdaric/lambda_runtime_feature_handler.py

```diff
@@ -0,0 +1,21 @@
+"""
+Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+"""
+import os
+from .lambda_runtime_features import feature_list
+
+
+class FeatureGate:
+    def __init__(self, execution_env=os.environ.get("AWS_EXECUTION_ENV")):
```

Comment:
Having `execution_env` as a particular env variable is a hardcode that would be beneficial to abstract from in a declarative manner. I.e. In [lambda_runtime_features.py](https://github.com/aws/aws-lambda-python-runtime-interface-client/pull/122/files#diff-d84d2cce0ac1caef4dcf5e1834bb74602d93109d833876f8de394beac429ccbd) for every feature specify source of the value to check against (be it env variable, file, sysfs/ioctl parameters etc)

Consider an example when `lambda_marshaller_ensure_ascii_false` feature should be not only gated by an env variable, but also by Python version obtained from `sys.version`. Another example is a feature for structured logging, implemented several commits before, it should also be easily pluggable into gating abstraction